lisp/ess-inf.el

```diff
@@ -241,30 +223,53 @@ This may be useful for debugging."
             (ess-wait-for-process proc)))
         inf-buf))))
 
-(defun inferior-ess--get-proc-buffer-create (proc-name)
-  "Get a process buffer, creating a new one if needed."
-  (let ((inf-name (funcall ess-gen-proc-buffer-name-function proc-name)))
-    (cond
-     ;; Try to use current buffer, if inferior-ess-mode but no process.
-     ;; Don't change existing buffer name in this case. It is very
-     ;; commong to restart the process in the same buffer.
-     ((and (not (comint-check-proc (current-buffer)))
-           (derived-mode-p 'inferior-ess-mode))
-      (current-buffer))
-     ;; Take the *R:N* buffer if already exists (and contains dead proc!)
-     ;; fixme: buffer name might have been changed, iterate over all
-     ;; inferior-ess buffers.
-     ((get-buffer inf-name))
-     ;; Pick up a transcript file
-     (ess-ask-about-transfile
-      (let ((transfilename (read-file-name
-                            "Use transcript file (default none):" nil "")))
-        (if (string= transfilename "")
-            (get-buffer-create inf-name)
-          (find-file-noselect (expand-file-name transfilename)))))
-     ;; Create a new buffer
-     (t
-      (get-buffer-create inf-name)))))
+(defun inferior-ess--get-proc-buffer-create (temp-dialect)
+  "Get a process buffer, creating a new one if needed.
```

Comment:
Checkdoc will complain about temp-dialect, can you mention it in the docstring please?

Reply:
oops, thank. I should get more familiar with checkdock.

Reply:
Flymake in Emacs 26+ uses it as one of its backends: `(add-hook 'emacs-lisp-mode-hook #'flymake-mode)`